Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -210,24 +219,28 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
     }
   }, [projectId]);
 
+  const usePrevious = (value) => {
+    const ref = useRef();
```

Comment:
I think this violates react rules on using hooks. They need to be used deterministically in the render of the component.

Reply:
Move usePrevious out of component